app/common/state/defaultBrowserState.js

```diff
@@ -8,5 +8,7 @@ const settings = require('../../../js/constants/settings')
 module.exports.shouldDisplayDialog = (state) => {
   return !getSetting(settings.IS_DEFAULT_BROWSER) &&
     !state.get('defaultBrowserCheckComplete') &&
-    getSetting(settings.CHECK_DEFAULT_ON_STARTUP)
+    getSetting(settings.CHECK_DEFAULT_ON_STARTUP) &&
+    process.env.NODE_ENV !== 'development' &&
+    process.env.NODE_ENV !== 'test'
```

Comment:
this looks good, but I think you probably want to add an option to override this check so we can test this without triggering it on every test. Maybe something like `process.env.BRAVE_FORCE_CHECK_DEFAULT_ON_STARTUP` which would run it regardless of the NODE_ENV or other settings